study/test/src/org/labkey/test/tests/study/StudyDatasetDomainTest.java

```
@@ -0,0 +1,237 @@
+        createCmd.setColumns(getColumnsWithDuplicateFields());
+        testForExpectedErrorMessage(createCmd, DUPLICATE_FIELD_ERROR_MSG);
+
+        int expectedErrorCount = 2;
+        assertEquals("Create Domain error count mismatch.", expectedErrorCount, NUM_ERRORS);
+    }
+
+    @Test
+    public void updateDomainErrorTest() throws IOException, CommandException
+    {
+        log("Create a study domain.");
+        DomainResponse domainResponse = createDomainWithGoodColumns(getGoodColumns());
+        long domainId = domainResponse.getDomainId();
+        String domainURI = domainResponse.getDomainURI();
+
+        log("Test for expected errors when saving an existing domain with a missing property.");
+        SaveDomainCommand saveCmd = new SaveDomainCommand(DOMAIN_KIND, STUDY_DATASET_NAME);
+        saveCmd.setDomainId((int)domainId);
+        saveCmd.setDomainURI(domainURI);
+        saveCmd.setSchemaName("study");
+        saveCmd.setColumns(getColumnWithBlankNameProperty());
+        testForExpectedErrorMessage(saveCmd, BLANK_PROPERTY_ERROR_MSG);
+
+        log("Test for expected errors when saving an existing domain with duplicate columns.");
+        saveCmd = new SaveDomainCommand(DOMAIN_KIND, STUDY_DATASET_NAME);
+        saveCmd.setDomainId((int)domainId);
+        saveCmd.setDomainURI(domainURI);
+        saveCmd.setSchemaName("study");
+        saveCmd.setColumns(getColumnsWithDuplicateFields());
+        testForExpectedErrorMessage(saveCmd, DUPLICATE_FIELD_ERROR_MSG);
+
+        int expectedErrorCount = 2;
+        assertEquals("Update Domain error count mismatch.", expectedErrorCount, NUM_ERRORS);
+    }
+
+    private void testForExpectedErrorMessage(DomainCommand cmd, String expectedErrorMsg)
+    {
+        try
+        {
+            cmd.execute(this.createDefaultConnection(false), getCurrentContainerPath());
+        }
+        catch (IOException | CommandException e)
```

Don't catch `IOException`. That exception would indicate that something much more serious went wrong. Just let it bubble up through the `@Tests`s

---

Good point! Done!